homeassistant/components/nmbs/sensor.py

```diff
@@ -227,6 +226,12 @@ def extra_state_attributes(self):
             ATTR_ATTRIBUTION: "https://api.irail.be/",
         }
 
+        if canceled != "1":
+            attrs["departure"] = f"In {departure} minutes"
+            attrs["departure_minutes"] = departure
+        else:
+            attrs["canceled"] = "1"
+
```

Comment:
Wouldn't it be better to always have a "canceled" attribute available that just contains "Yes" or "No", no matter the canceled state? That would probably make it easier to use in automations.

So basically if `canceled` == "1": set "Canceled" as the departure string and set the canceled attribute to True (if it can be a boolean, but I can't immediately see why not). I'm not sure what `departure_minutes` would best be set at, but I guess 0 would be fine?
For all other values of `canceled`, the canceled can be set as False and the other attributes can be set as before.

Reply:
I'd go with a boolean that is always present. We're moving to always keeping the key present in the state attributes.

Would it work to extract the feature to a separate binary sensor for each departure that is tracked?

Reply:
Should I keep departure and departure_minutes too then? Set them respectively to "Canceled" and the original departure time. This is the actual behaviour, it makes sense IMO to have the original time but a cancelled state. Or infinite for the minutes?

I wouldn't recommend a separate binary sensor. I use a modified version of this sensor that tracks the next N trains and not just the next one, which I think should be the next planned feature. When I look at HA I usually look at the next train or the one after to decide if I should go now or should wait. If there are even more sensors it becomes a bit messy...

Reply:
I can agree with that as well. Certainly `departure` and maybe also `departure_minutes` should still have some value in the case of a cancellation. If infinite time is a valid value, I think it would make sense in this case.

An option to ignore canceled trains would indeed make sense from a user perspective. I'd honestly almost even prefer that to be the default behavior.